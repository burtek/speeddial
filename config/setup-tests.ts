import '@testing-library/jest-dom';
import { createSerializer } from '@emotion/jest';
import i18n from 'i18next';
import 'jest-extended';
import 'jest-extended/all';
import { initReactI18next } from 'react-i18next';


expect.addSnapshotSerializer(createSerializer());

await i18n
    .use(initReactI18next)
    .init({
        lng: 'dev',
        interpolation: { escapeValue: false }
    });
