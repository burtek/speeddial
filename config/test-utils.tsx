import type { RenderOptions } from '@testing-library/react';
import { render } from '@testing-library/react';
import type { FC, PropsWithChildren, ReactElement } from 'react';
import { Provider as StoreProvider } from 'react-redux';

import { store } from '@@data/index';
import { ThemedApp } from 'src/theme/themed-app';

const AllTheProviders: FC<PropsWithChildren> = ({ children }) => (
    <StoreProvider store={store}>
        <ThemedApp>
            {children}
        </ThemedApp>
    </StoreProvider>
);

const customRender = (
    ui: ReactElement,
    options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
// eslint-disable-next-line import/export
export * from '@testing-library/react';

// override render method
// eslint-disable-next-line import/export
export { customRender as render };
