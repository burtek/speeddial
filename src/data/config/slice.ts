import type { PayloadAction } from '@reduxjs/toolkit';
import { createSlice } from '@reduxjs/toolkit';

export type ThemeMode = 'dark' | 'light';

export const initialState = () => ({
    modalOpen: false,
    themeMode: null as ThemeMode | null
});

export const { actions, name: sliceName, reducer } = createSlice({
    name: 'config',
    initialState,
    reducers: {
        openModal(state) {
            state.modalOpen = true;
        },
        closeModal(state) {
            state.modalOpen = false;
        },

        setThemeMode(state, { payload }: PayloadAction<{ mode: ThemeMode | null }>) {
            state.themeMode = payload.mode;
        }
    }
});
