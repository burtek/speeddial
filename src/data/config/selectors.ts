import { createSelector } from '@reduxjs/toolkit';

import type { RootState } from '..';

import { sliceName } from './slice';


export const getConfig = (state: RootState) => state[sliceName];

export const getThemeMode = createSelector(getConfig, s => s.themeMode);

export const getIsDialogOpen = createSelector(getConfig, s => s.modalOpen);
