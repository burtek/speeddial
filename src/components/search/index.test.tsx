import { render } from 'config/test-utils';

import { SearchBar } from './index';


describe('search-bar', () => {
    it('should match snapshot', () => {
        const { container } = render(<SearchBar />);

        expect(container).toMatchSnapshot();
    });
});
