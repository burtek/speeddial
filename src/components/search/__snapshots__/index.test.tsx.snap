// Vitest Snapshot v1

exports[`search-bar > should match snapshot 1`] = `
.emotion-0 {
  background-color: #fff;
  color: rgba(0, 0, 0, 0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  box-shadow: 0px 3px 3px -2px rgba(0,0,0,0.2),0px 3px 4px 0px rgba(0,0,0,0.14),0px 1px 8px 0px rgba(0,0,0,0.12);
  padding: 2vw;
}

<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 emotion-0"
  >
    <form
      action="https://www.google.com/search"
      method="get"
      style="display: flex;"
    >
      <input
        name="q"
        placeholder="search.placeholder"
        style="flex: 1; padding: 0.3vw 0.5vw; border-radius: 5px;"
        type="text"
      />
    </form>
  </div>
</div>
`;
